src/GitStatusCache/src/Git.cpp

```diff
@@ -270,6 +271,20 @@ bool Git::GetRefStatus(Git::Status& status, UniqueGitRepository& repository)
 		Log("Git.GetRefStatus.NoUpstream", Severity::Spam)

```

Comment:
Hrm. I think I like it logged with the current behavior since we don't populate Upstream in this case. If we change that behavior to report the old Upstream details, I think your suggested adjustment makes sense.